src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -268,9 +274,12 @@ private function createAvailableMessagesQueryBuilder(): QueryBuilder
             ->andWhere('m.available_at <= :now')
             ->andWhere('m.queue_name = :queue_name')
             ->setParameters([
-                ':now' => self::formatDateTime($now),
+                ':now' => $now,
                 ':queue_name' => $this->configuration['queue_name'],
-                ':redeliver_limit' => self::formatDateTime($redeliverLimit),
+                ':redeliver_limit' => $redeliverLimit,
+            ], [
+                ':now' => Type::getType($this->getDateTimeType()),
```

Comment:
do you really need to pass the type instance here with `Type::getType`? I don't think so. Passing the constant value should be enough.